accounts/service.go

```diff
@@ -563,6 +655,24 @@ func (s *InterceptorService) paymentUpdate(hash lntypes.Hash,
 	s.Lock()
 	defer s.Unlock()
 
+	// As this function updates account balances, we ensure that the service
+	// is running before we execute it.
+	if err := s.requireRunning(); err != nil {
+		// We will update the payment on next startup instead, once the
+		// error that caused the service to stop has been resolved.
+		return false, err
+	}
+	defer func() {
+		// We want to ensure that we disable the service if we error
+		// when making the payment update. We need to do this while
+		// still holding the lock to ensure that other waiting threads
+		// to grab a lock will get informed immediately that the service
+		// is disabled.
+		if returnErr != nil {
+			s.disable()
+		}
```

Comment:
related to my previous suggestion: perhaps instead there should just be one place to send critical errors and then that can be in-charge of updating the account-service status which will then ensure that no more requests come through